openfasoc/generators/gdsfactory-gen/glayout/pdk/util/port_utils.py

```diff
@@ -103,6 +103,19 @@ def rename_ports_by_list(custom_comp: Component, replace_list: list[tuple[str,st
     return rename_component_ports(custom_comp, rename_func)
 
 
+def remove_ports_with_prefix(custom_comp: Component, prefix: str) -> Component:
```

Comment:
Replace with list comprehension? I actually find that a lot of methods in this and other files can be replaced with this more efficient implementation.

```python
return [port for port in custom_comp.ports if not port.startswith(prefix)]
```

Reply:
Actually I think it has to be written to the object in `custom_comp` (oops):
```python
remove_list = [prt for prt in custom_comp.ports.keys() if prt.startswith(prefix)]
[custom_comp.ports.pop(prt) for prt in remove_list]
return custom_comp
```